File: WorkIt/app/src/main/java/prot3ct/workit/data/local/UserSession.java
package prot3ct.workit.data.local;

import android.app.backup.FullBackupDataOutput;
import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import prot3ct.workit.data.local.base.UserSessionContract;

public class UserSession implements UserSessionContract{
    private final SharedPreferences sharedPreferences;

    public UserSession(Context context) {
        this.sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
    }

    @Override
    public String getAccessToken() {
        return sharedPreferences.getString("accessToken", null);
    }

    @Override
    public String getFullName() {
        return sharedPreferences.getString("fullName", null);
    }

    @Override
    public String getEmail() {
        return sharedPreferences.getString("email", null);
    }

    @Override
    public int getId() {
        return sharedPreferences.getInt("id", 0);
    }

    @Override
    public void setEmail(String email) {
        sharedPreferences.edit().putString("email", email).apply();
    }

    @Override
    public void setFullName(String fullName) {
        sharedPreferences.edit().putString("fullName", fullName).apply();
    }

    @Override
    public void setAccessToken(String accessToken) {
        sharedPreferences.edit().putString("accessToken", accessToken).apply();
    }

    @Override
    public void setId(int id) {
        sharedPreferences.edit().putInt("id", id).apply();
    }

    @Override
    public boolean isUserLoggedIn() {
        return getEmail() != null;
    }

    @Override
    public void clearSession() {
        setId(0);
        setFullName(null);
        setEmail(null);
        setAccessToken(null);
    }
}

File: WorkIt/app/src/main/java/prot3ct/workit/data/local/UserSession.kt
package prot3ct.workit.data.local

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager

class UserSession(context: Context) {
    private val sharedPreferences: SharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)

    var id: Int
        get() = sharedPreferences.getInt("id", 0)
        set(newId) {
            sharedPreferences.edit().putInt("id", newId).apply()
        }

    var accessToken: String?
        get() = sharedPreferences.getString("accessToken", null)
        set(accessToken) {
            sharedPreferences.edit().putString("accessToken", accessToken).apply()
        }

    var fullName: String?
        get() = sharedPreferences.getString("fullName", null)
        set(fullName) {
            sharedPreferences.edit().putString("fullName", fullName).apply()
        }

    var email: String?
        get() = sharedPreferences.getString("email", null)
        set(email) {
            sharedPreferences.edit().putString("email", email).apply()
        }

    var loggedInUserId: Int
        get() = sharedPreferences.getInt("id", 0)
        set(id) {
            sharedPreferences.edit().putInt("id", id).apply()
        }

    val isLoggedIn: Boolean
        get() = email != null

    fun clearSession() {
        id = 0
        fullName = null
        email = null
        accessToken = null
    }
}